### PR TITLE
[EXPLORER] 'Run' (Win+R) should set the proper current directory

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

'Start' Menu -> 'Run' should set the proper current directory for new process.

JIRA issue: [CORE-14583](https://jira.reactos.org/browse/CORE-14583)

## Proposed changes

The target source is "base/shell/explorer/traywnd.cpp".

- Set the proper directory to the 3rd parameter of RunFileDlg call.

### CODE REVIEW

File: base/shell/explorer/traywnd.cpp [hunk 1]
```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
...
-        RunFileDlg(hwnd, NULL, NULL, NULL, NULL, RFF_CALCDIRECTORY);
+        WCHAR szHomeDrive[MAX_PATH], szHomePath[MAX_PATH];
+        GetEnvironmentVariableW(L"HOMEDRIVE", szHomeDrive, _countof(szHomeDrive));
+        GetEnvironmentVariableW(L"HOMEPATH", szHomePath, _countof(szHomePath));
+
+        WCHAR szDefaultDir[MAX_PATH];
+        StringCchCopyW(szDefaultDir, _countof(szDefaultDir), szHomeDrive);
+        StringCchCatW(szDefaultDir, _countof(szDefaultDir), szHomePath);
+
+        RunFileDlg(hwnd, NULL, szDefaultDir, NULL, NULL, RFF_CALCDIRECTORY);
```
Comment (HBelusca) [inline base/shell/explorer/traywnd.cpp]: Interesting observation:
- explorer.exe indeed uses:
```c
RunFileDlg(hWnd, NULL, "C:\Users\username", "Execute", NULL, RFF_CALCDIRECTORY)
```
- while taskmgr, from the "File - New Task" menu item, uses:
```c
RunFileDlg(hWnd, ptr1, "C:\Windows\system32", "Create a new task", "Dialog description", RFF_CALCDIRECTORY | RFF_NOSEPARATEMEM | 64)
```

Comment (katahiromz) [reply base/shell/explorer/traywnd.cpp]: Shall we change "File - New Task" menu item action?

File: base/shell/explorer/traywnd.cpp [hunk 2]
```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
 
         m_RunFileDlgOwner = hwnd;
 
-        RunFileDlg(hwnd, NULL, NULL, NULL, NULL, RFF_CALCDIRECTORY);
+        WCHAR szHomeDrive[MAX_PATH], szHomePath[MAX_PATH];
+        GetEnvironmentVariableW(L"HOMEDRIVE", szHomeDrive, _countof(szHomeDrive));
+        GetEnvironmentVariableW(L"HOMEPATH", szHomePath, _countof(szHomePath));
```
Comment (ThFabba) [inline base/shell/explorer/traywnd.cpp]: Why not SHGetFolderPath(... CSIDL_PROFILE...)?

Comment (katahiromz) [reply base/shell/explorer/traywnd.cpp]: According to API Monitor investigation, 2k3 is doing so.

File: base/shell/explorer/traywnd.cpp [hunk 3]
```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
...
+
+        WCHAR szDefaultDir[MAX_PATH];
+        StringCchCopyW(szDefaultDir, _countof(szDefaultDir), szHomeDrive);
+        StringCchCatW(szDefaultDir, _countof(szDefaultDir), szHomePath);
```
Comment (HBelusca) [inline base/shell/explorer/traywnd.cpp]: You can use CStringW variables instead:
- one that will contain the default directory "szDefaultDir", which is first initialized by a call to CStringW::GetEnvironmentVariable to initialize it with HOMEDRIVE;
- a second one that is initialized (again with CStringW::GetEnvironmentVariable) with HOMEPATH,
- then you concatenate both of these back into szDefaultDir (szDefaultDir += szHomePath),
- and you use (LPCWSTR)szDefaultDir as the 3rd parameter in RunFileDlg.

The resulting code should be cleaner and in the spirit of C++ used in shell/explorer.

Comment (katahiromz) [reply base/shell/explorer/traywnd.cpp]: Thank you for reviewing. Done.

**Pull request merged**